server/portal/apps/workbench/templates/portal/apps/workbench/index.j2

```diff
@@ -12,6 +12,7 @@
   <div id="react-root"></div>
 {% endblock %}
 {% block scripts %}
+  <script> window.__INITIAL_SETUP_COMPLETE__ = {{ setup_complete|yesno:"true,false" }}; </script>
```

Comment:
Does this speed up the page render? Or is it for something else

Reply:
This is to avoid waiting on an initial request to find out the user's `setup_complete`.

So basically, we're wanting to know the state of setup_compete **immediately**. This makes frontend code simpler (?) as we immediately know what routes are available to the user.  And so we can allow them immediately to someplace like /workspace/applications (on an initial load) instead of having to show a spinner while we make a request to the backend to see if their setup_complete is true.

not certain if this use of adding something to the `window` in the template is the best way to pass initial values to our react app.  so any feedback/direction there is appreciated.

Reply:
This came about as we moved from having `/onboarding` to `/workbench/onboarding`.

Previously, when we had the  `/onboarding`, the backend to redirect you from `/workbench/something` to `/onboarding` if setup_complete was False. 

Now, we have placed onboarding at `/workbench/onboarding` and  we are checking setup_complete **on the client side** to determine if users can get to `/workbench/something` or need to go instead to `/workbench/onboarding`. If needed, I can provide more details about what motivated us to make that switch.

Reply:
Is then the response from the backend with the `setupComplete` value needed, if we get that from `window`?

Reply:
Yes, because the state can change, so we will need useSelector and redux. 

I think this is a general stop gap for not having a loading page or spinner for the entire single page app. If we had such a thing, we could do this purely in redux with something to the effect of setting the initialState of setupComplete to a "unknown" or something, and then after the initial dispatches from the useEffect from the workbench completes, we will have an actual value for the flag. We could even do something more general, like having a saga and state that keeps track of the required sagas that need to finish on initial loading before displaying the workbench. But since we don't have a loading page for now, this is the hack we have.

Reply:
Yeah, we can remove the stop gap and do something more elegant than the `window.__INITIAL_SETUP_COMPLETE__` (as Joon Yee outlines above). If so, I would prefer a follow up Jira issue/PR as we would require a front end design as what the loading page looks like.

Reply:
👍  Was a follow up ticket made for this?